src/libstd/sys/unix/weak.rs

```diff
@@ -18,15 +18,15 @@
 
 use libc;
 
-use ffi::CString;
+use ffi::CStr;
 use marker;
 use mem;
 use sync::atomic::{AtomicUsize, Ordering};
 
 macro_rules! weak {
     (fn $name:ident($($t:ty),*) -> $ret:ty) => (
         static $name: ::sys::weak::Weak<unsafe extern fn($($t),*) -> $ret> =
-            ::sys::weak::Weak::new(stringify!($name));
+            ::sys::weak::Weak::new(concat!(stringify!($name), '\0'));
     )
 }
```

Comment:
Shouldn't *something* about this abstraction be unsafe?

Reply:
Sure, I think it would make sense for the constructor to be `unsafe`, which the macro can hide knowing this is `unsafe fn`. All uses do go through this macro already though.